lib/activerecord/delay_touching.rb

```diff
@@ -6,9 +6,9 @@ module DelayTouching
     extend ActiveSupport::Concern
 
     # Override ActiveRecord::Base#touch.
-    def touch(name = nil)
+    def touch(*names)
       if self.class.delay_touching? && !try(:no_touching?)
-        DelayTouching.add_record(self, name)
+        DelayTouching.add_record(self, names)
```

Comment:
The state class is responsible for storing this data, and I think moving the logic up to the delay_touching module is  misplacing this concern; `state` should ensure that the data it's given is valid. Otherwise, I'll also have to add logic in DelayTouching.touch_records to ensure that the data I get back out of state is valid. This logic would also have to be repeated in any other class in the future that might use the state class.
I definitely feel strongly that the logic for ensuring properly formed data belongs in the state class.